### PR TITLE
🐛 Use gingko cli and timeout

### PR DESCRIPTION
**Pull request by mboukhalfa**
**What this PR does / why we need it**:
Remove time command when running e2e and ginkgo cli instead of go test

### CODE REVIEW

**Review by lentzi90 — COMMENTED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -235,6 +239,10 @@ $(SETUP_ENVTEST): $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR); go get sigs.k8s.io/controller-runtime/tools/setup-envtest; \
 	go build -tags=tools -o $(BIN_DIR)/setup-envtest sigs.k8s.io/controller-runtime/tools/setup-envtest
 
+$(GINKGO): $(TOOLS_DIR)/go.mod
+	cd $(TOOLS_DIR); go get github.com/onsi/ginkgo/v2/ginkgo@v2.6.0
```
Comment (tuminoid) [inline Makefile:243]: `cd; cmd` pattern should always be written as `cd && cmd`, so if the cd fails it doesn't continue execution in wrong path.

File: Makefile [hunk 2]
```diff
@@ -162,27 +163,30 @@ cluster-templates: $(KUSTOMIZE) ## Generate cluster templates
...
-	cd test; \
-	time go test -v -timeout 24h --tags=e2e ./e2e/... -args \
-		--ginkgo.timeout=6h --ginkgo.v --ginkgo.trace --ginkgo.show-node-events --ginkgo.no-color=$(GINKGO_NOCOLOR) \
-		--ginkgo.junit-report="junit.e2e_suite.1.xml" \
-		--ginkgo.focus="$(GINKGO_FOCUS)" $(_SKIP_ARGS) \
+
+	$(GINKGO) --timeout=$(GINKGO_TIMEOUT) -v --trace --tags=e2e  \
+		--show-node-events --no-color=$(GINKGO_NOCOLOR) \
+		--junit-report="junit.e2e_suite.1.xml" \
+		--focus="$(GINKGO_FOCUS)" $(_SKIP_ARGS) $(ROOT_DIR)/$(TEST_DIR)/e2e/ -- \
```
Comment (tuminoid) [inline Makefile:183]: ROOT_DIR may contain spaces so this breaks - needs quotes

**Review by tuminoid — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (tuminoid) [reply Makefile:183]: Same with $GINKGO as its using $ROOT_DIR earlier

Comment (mboukhalfa) [reply Makefile]: Thanks fixed

File: Makefile (hunk 1 above)
Comment (mboukhalfa) [reply Makefile]: fixed

**Comment by mquhuy** [comment]
I noticed that ginkgo cli was used in the past, then got changed to `go test`. Could you clarify what caused that, and why do we change it back to ginkgo cli now?

**Comment by mboukhalfa** [comment]
> I noticed that ginkgo cli was used in the past, then got changed to `go test`. Could you clarify what caused that, and why do we change it back to ginkgo cli now?

Hmm I was not aware that `ginkgo` was used before and it seems changed to `go test` just to avoid installing `ginkgo` though my motivation to use g`inkgo cli` here is to avoid changing directory to run e2e tests and clean the arguments i.e avoid using `--ginkgo` prefix before each arg also I noted on ginkgo docs that ginkgo cli must be needed for certain feature like parallel which still not needed in our case

**Comment by mquhuy** [comment]
> > I noticed that ginkgo cli was used in the past, then got changed to `go test`. Could you clarify what caused that, and why do we change it back to ginkgo cli now?
> 
> Hmm I was not aware that `ginkgo` was used before and it seems changed to `go test` just to avoid installing `ginkgo` though my motivation to use `ginkgo cli` here is to avoid changing directory to run e2e tests and clean the arguments i.e avoid using `--ginkgo` prefix before each arg also I noted on ginkgo docs that ginkgo cli must be needed for certain feature like parallel which still not needed in our case

Thanks. I think that's reasonable, but cannot make sure since it was purposefully changed to `go test` before. Here is the PR link: https://github.com/metal3-io/cluster-api-provider-metal3/pull/194

**Review by tuminoid — APPROVED** [review]

**Comment by mboukhalfa** [comment]
/assign @kashifest 
cc @furkatgofurov7

**Pull request merged**